core/src/main/scala/cats/implicits.scala

```diff
@@ -7,6 +7,7 @@ object implicits
     with syntax.AllSyntaxBinCompat2
     with syntax.AllSyntaxBinCompat3
     with syntax.AllSyntaxBinCompat4
+    with syntax.AllSyntaxBinCompat5
```

Comment:
`AllSyntaxBinCompat4` was introduced after the last release.  You can simply let it extends your `FoldableSyntaxBinCompat1`